crates/stark/src/prover.rs

```diff
@@ -423,6 +428,15 @@ where
                             let permutation_trace_on_quotient_domains = pcs
                                 .get_evaluations_on_domain(&permutation_data, i, *quotient_domain)
                                 .to_row_major_matrix();
+
+                            let chip_num_constraints =
+                                pk.constraints_map.get(&chips[i].name()).unwrap();
+
```

Comment:
I think adding a comment on why you are doing a `take` and a `reverse` here would be good.  It's not obvious from the code of the reason.